--- initramfs-tools/hook-functions.orig	2013-10-20 19:18:52.279496090 +0200
+++ initramfs-tools/hook-functions	2013-10-20 19:57:25.009399034 +0200
@@ -59,7 +59,7 @@
 		return
 	fi
 
-	modprobe --all --set-version="${version}" --ignore-install --quiet --show-depends "$@" |
+	/sbin/modprobe --all --set-version="${version}" --ignore-install --quiet --show-depends "$@" |
 	while read prefix kmod options ; do
 		if [ "${prefix}" != "insmod" ]; then
 			continue
diff -ur initramfs-tools-0.93/mkinitramfs initramfs-tools-0.93-nobb/mkinitramfs
--- initramfs-tools-0.93/mkinitramfs	2009-02-23 17:10:14.000000000 +0100
+++ initramfs-tools-0.93-nobb/mkinitramfs	2009-02-23 17:11:07.000000000 +0100
@@ -135,7 +135,7 @@
 	exit 1
 fi
 if [ ! -e "${MODULESDIR}/modules.dep" ]; then
-	depmod ${version}
+	/sbin/depmod ${version}
 fi
 
 DESTDIR="$(mktemp -t -d mkinitramfs_XXXXXX)" || exit 1
--- initramfs-tools/scripts/functions.orig	2013-09-12 14:39:58.000000000 +0200
+++ initramfs-tools/scripts/functions	2013-10-20 19:59:34.126060282 +0200
@@ -44,14 +44,14 @@
 		sleep ${panic}
 		reboot
 	fi
-	modprobe -v i8042 || true
-	modprobe -v atkbd || true
-	modprobe -v ehci-pci || true
-	modprobe -v ehci-orion || true
-	modprobe -v ehci-hcd || true
-	modprobe -v uhci-hcd || true
-	modprobe -v ohci-hcd || true
-	modprobe -v usbhid || true
+	/sbin/modprobe -v i8042 || true
+	/sbin/modprobe -v atkbd || true
+	/sbin/modprobe -v ehci-pci || true
+	/sbin/modprobe -v ehci-orion || true
+	/sbin/modprobe -v ehci-hcd || true
+	/sbin/modprobe -v uhci-hcd || true
+	/sbin/modprobe -v ohci-hcd || true
+	/sbin/modprobe -v usbhid || true
 	REASON="$@" PS1='(initramfs) ' /bin/sh -i </dev/console >/dev/console 2>&1
 }
 
@@ -261,7 +261,7 @@
 			if [ "$com" = "#" ]; then
 				continue
 			fi
-			modprobe $m
+			/sbin/modprobe $m
 		done
 	fi
 }
--- initramfs-tools/scripts/init-top/all_generic_ide.orig	2013-10-20 20:00:29.019391312 +0200
+++ initramfs-tools/scripts/init-top/all_generic_ide	2013-10-20 20:04:39.002714154 +0200
@@ -16,11 +16,11 @@
 for x in $(cat /proc/cmdline); do
 	case ${x} in
 	all_generic_ide)
-		modprobe ata_generic all_generic_ide=1
+		/sbin/modprobe ata_generic all_generic_ide=1
 		;;
 	all_generic_ide=*)
 		if [ ${x#all_generic_ide=} ]; then
-			modprobe ata_generic all_generic_ide=1
+			/sbin/modprobe ata_generic all_generic_ide=1
 		fi
 		;;
 	esac
--- initramfs-tools/docs/framebuffer.orig	2013-07-23 19:19:27.000000000 +0200
+++ initramfs-tools/docs/framebuffer	2013-10-20 20:07:01.579374837 +0200
@@ -85,7 +85,7 @@
 	;;
 intelfb|i810fb|i915)
 	# Needs AGP driver loaded
-	modprobe intel-agp
+	/sbin/modprobe intel-agp
 	;;
 uvesafb)
 	# v86d requires /dev/zero and dev/mem, but udev haven't been started yet
@@ -98,8 +98,8 @@
 
 if [ -n "${FB}" ]; then
 	unset MODPROBE_OPTIONS
-	modprobe -q fbcon
-	modprobe -q ${FB} ${OPTS}
+	/sbin/modprobe -q fbcon
+	/sbin/modprobe -q ${FB} ${OPTS}
 fi
 
 if [ -e /proc/fb ]; then
diff -ur initramfs-tools-0.93/scripts/local initramfs-tools-0.93-nobb/scripts/local
--- initramfs-tools-0.93/scripts/local	2009-02-17 15:25:56.000000000 +0100
+++ initramfs-tools-0.93-nobb/scripts/local	2009-02-23 17:13:21.000000000 +0100
@@ -123,7 +123,7 @@
 	fi
 
 	# FIXME This has no error checking
-	modprobe ${FSTYPE}
+	/sbin/modprobe ${FSTYPE}
 
 	# FIXME This has no error checking
 	# Mount root
diff -ur initramfs-tools-0.93/scripts/nfs initramfs-tools-0.93-nobb/scripts/nfs
--- initramfs-tools-0.93/scripts/nfs	2008-08-13 15:23:44.000000000 +0200
+++ initramfs-tools-0.93-nobb/scripts/nfs	2009-02-23 17:13:31.000000000 +0100
@@ -55,9 +55,9 @@
 	run_scripts /scripts/nfs-top
 	[ "$quiet" != "y" ] && log_end_msg
 
-	modprobe nfs
+	/sbin/modprobe nfs
 	# For DHCP
-	modprobe af_packet
+	/sbin/modprobe af_packet
 
 	wait_for_udev 10
 
