--- initramfs-tools/hook-functions.orig	2013-10-20 20:07:37.766039986 +0200
+++ initramfs-tools/hook-functions	2013-10-20 20:24:00.679332070 +0200
@@ -93,9 +93,9 @@
 				fi
 
 				kmod_modname="${kmod##*/}"
-				kmod_modname="${kmod_modname%.ko}"
+				kmod_modname="${kmod_modname%.ko*}"
 				if grep -q "^$kmod_modname\\>" /proc/modules "${CONFDIR}/modules"; then
-					echo "W: Possible missing firmware /lib/firmware/${firmware} for module $(basename ${kmod} .ko)" >&2
+					echo "W: Possible missing firmware /lib/firmware/${firmware} for module $kmod_modname" >&2
 				fi
 				continue
 			fi
@@ -187,8 +187,8 @@
 		exclude="${exclude:-} -name $1 -prune -o "
 		shift
 	done
-	for kmod in $(find "${MODULESDIR}/${dir}" ${exclude:-} -name '*.ko' -printf '%f\n'); do
-		modules="$modules ${kmod%.ko}"
+	for kmod in $(find "${MODULESDIR}/${dir}" ${exclude:-} \( -name '*.ko' -o -name '*.ko.gz' \) -printf '%f\n'); do
+		modules="$modules ${kmod%.ko*}"
 	done
 	manual_add_modules $modules
 }
