--- initramfs-tools/hooks/klibc.orig	2013-09-12 15:20:05.000000000 +0200
+++ initramfs-tools/hooks/klibc	2013-10-20 20:54:57.609254142 +0200
@@ -16,16 +16,17 @@
 esac
 
 # klibc
-cp -pnL /usr/lib/klibc/bin/* ${DESTDIR}/bin
-cp -pL /lib/klibc-*.so ${DESTDIR}/lib
+mkdir -p ${DESTDIR}/__KLIBCDIR__
+cp -pnL /usr/__KLIBCDIR__/klibc/bin/* ${DESTDIR}/bin
+cp -pL /__KLIBCDIR__/klibc-*.so ${DESTDIR}/__KLIBCDIR__
 rm -f ${DESTDIR}/bin/kinit* ${DESTDIR}/bin/zcat
 if [ "${BUSYBOX}" = "n" ] || [ ! -e ${BUSYBOXDIR}/busybox ]; then
 	if [ -e ${DESTDIR}/bin/sh.shared ]; then
 		# Some platforms build a shared klibc/sh:
 		mv ${DESTDIR}/bin/sh.shared ${DESTDIR}/bin/sh
-	elif [ -e /usr/lib/klibc/bin/sh ]; then
+	elif [ -e /usr/__KLIBCDIR__/klibc/bin/sh ]; then
 		# Others build a static version instead:
-		cp -pL /usr/lib/klibc/bin/sh ${DESTDIR}/bin/sh
+		cp -pL /usr/__KLIBCDIR__/klibc/bin/sh ${DESTDIR}/bin/sh
 	fi
 else
 	# Nobody wanted it, so save a tiny bit of space:
