--- initramfs-tools/conf/initramfs.conf~	2009-03-19 21:16:14.000000000 +0100
+++ initramfs-tools/conf/initramfs.conf	2010-01-20 17:29:01.818021234 +0100
@@ -28,6 +28,14 @@
 BUSYBOX=y
 
 #
+# INITRDTOOLS: [ y | n ]
+#
+# Use dedicated programs for init(rd|ramfs) usage if available
+#
+
+INITRDTOOLS=y
+
+#
 # KEYMAP: [ y | n ]
 #
 # Load a keymap during the initramfs stage.
--- initramfs-tools/mkinitramfs.orig	2013-10-20 20:28:07.915988361 +0200
+++ initramfs-tools/mkinitramfs	2013-10-20 20:37:00.889299328 +0200
@@ -10,6 +10,7 @@
 test -e /bin/busybox && BUSYBOXDIR=/bin
 test -e /usr/lib/initramfs-tools/bin/busybox && BUSYBOXDIR=/usr/lib/initramfs-tools/bin
 export BUSYBOXDIR
+INITRDDIR="/usr/lib/initrd"
 
 OPTIONS=`getopt -o c:d:ko:r:v -n "$0" -- "$@"`
 
@@ -190,6 +191,8 @@
 export KEYMAP
 export MODULES
 export BUSYBOX
+export INITRDDIR
+export INITRDTOOLS
 
 # Private, used by 'catenate_cpiogz'.
 export __TMPCPIOGZ
--- initramfs-tools/hook-functions.orig	2013-10-20 20:38:05.389296621 +0200
+++ initramfs-tools/hook-functions	2013-10-20 20:40:48.235956453 +0200
@@ -123,6 +123,17 @@
 	src="${1}"
 	target="${2:-$1}"
 
+	if [ "${INITRDTOOLS}" = "y" ] || [ "${INITRDTOOLS}" = "Y" ]; then
+		x="$src"
+		while [[ "$x" = */* ]]; do
+			[ -e "${INITRDDIR}/${x}" ] && break
+			x=${x#*/}
+		done
+		if [ -e "${INITRDDIR}/${x}" ]; then
+			src="${INITRDDIR}/${x}"
+		fi
+	fi
+		
 	[ -f "${src}" ] || return 1
 
 	if [ -d "${DESTDIR}/${target}" ]; then
